gluten-flink/ut/src/test/java/org/apache/gluten/streaming/api/operators/GlutenStreamFilterTest.java

```diff
@@ -0,0 +1,368 @@
+            GenericRowData.of(4, StringData.fromString("David"), 15));
+  }
+
+  @Test
+  public void testGreaterThanFilter() throws Exception {
+    List<RowData> expectedOutput =
+        Arrays.asList(
+            GenericRowData.of(1, StringData.fromString("Alice"), 20),
+            GenericRowData.of(3, StringData.fromString("Charlie"), 25));
+
+    testComparisonFilter(SqlStdOperatorTable.GREATER_THAN, 18, expectedOutput);
+  }
+
+  @Test
+  public void testLessThanFilter() throws Exception {
+    List<RowData> expectedOutput =
+        Arrays.asList(
+            GenericRowData.of(2, StringData.fromString("Bob"), 17),
+            GenericRowData.of(4, StringData.fromString("David"), 15));
+
+    testComparisonFilter(SqlStdOperatorTable.LESS_THAN, 20, expectedOutput);
+  }
+
+  @Test
+  public void testEqualToFilter() throws Exception {
+    List<RowData> expectedOutput =
+        Arrays.asList(GenericRowData.of(1, StringData.fromString("Alice"), 20));
+
+    testComparisonFilter(SqlStdOperatorTable.EQUALS, 20, expectedOutput);
+  }
+
+  @Test
+  @Disabled
+  public void testGreaterThanOrEqualFilter() throws Exception {
+    List<RowData> expectedOutput =
+        Arrays.asList(
+            GenericRowData.of(1, StringData.fromString("Alice"), 20),
+            GenericRowData.of(3, StringData.fromString("Charlie"), 25));
+
+    testComparisonFilter(SqlStdOperatorTable.GREATER_THAN_OR_EQUAL, 20, expectedOutput);
+  }
+
+  @Test
+  @Disabled
+  public void testLessThanOrEqualFilter() throws Exception {
+    List<RowData> expectedOutput =
+        Arrays.asList(
+            GenericRowData.of(1, StringData.fromString("Alice"), 20),
+            GenericRowData.of(2, StringData.fromString("Bob"), 17),
+            GenericRowData.of(4, StringData.fromString("David"), 15));
+
+    testComparisonFilter(SqlStdOperatorTable.LESS_THAN_OR_EQUAL, 20, expectedOutput);
+  }
+
+  @Test
+  @Disabled
+  public void testNotEqualFilter() throws Exception {
+    List<RowData> expectedOutput =
+        Arrays.asList(
+            GenericRowData.of(2, StringData.fromString("Bob"), 17),
+            GenericRowData.of(3, StringData.fromString("Charlie"), 25),
+            GenericRowData.of(4, StringData.fromString("David"), 15));
+
+    testComparisonFilter(SqlStdOperatorTable.NOT_EQUALS, 20, expectedOutput);
+  }
+
+  @Test
+  public void testStringEqualFilter() throws Exception {
+
+    RexNode nameFieldRef =
+        rexBuilder.makeInputRef(typeFactory.createSqlType(SqlTypeName.VARCHAR), 1);
+    RexNode name =
+        rexBuilder.makeLiteral("Alice", typeFactory.createSqlType(SqlTypeName.VARCHAR), false);
+    RexNode flinkFilterCondition =
+        rexBuilder.makeCall(SqlStdOperatorTable.EQUALS, nameFieldRef, name);
+
+    List<RowData> expectedOutput =
+        Arrays.asList(GenericRowData.of(1, StringData.fromString("Alice"), 20));
+
+    testFilterWithCondition(flinkFilterCondition, expectedOutput);
```

Comment:
I note some function calls this to filter name and some calls testComparisonFilter to filter age. It may be better to use a unified util function called `testFilter` with `RexNode flinkFilterCondition` as its argument. The `flinkFilterCondition` is constructed at caller side.